src/Symfony/Component/Form/Extension/Core/Type/DayOfWeekType.php

```diff
@@ -0,0 +1,117 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Form\Extension\Core\Type;
+
+use Symfony\Component\Form\AbstractType;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+use Symfony\Component\OptionsResolver\Options;
+use Symfony\Component\Form\Extension\Core\ChoiceList\SimpleChoiceList;
+use Symfony\Component\Form\Extension\Core\DataTransformer\DayOfWeekTransformer;
+
+/**
+ * Choice between all days of the week
+ * Values are <1-7>, 1 = monday, 7 = sunday, ISO-8601
+ * Labels are the localized names (Monday-Sunday)
+ * The first day of the week is governed by the Locale
+ *
+ * @author Sebastien Lavoie <seb@wemakecustom.com>
+ */
+class DayOfWeekType extends AbstractType
+{
+    /**
+     * IntlCalendar::DOW_* constants are not compatible with ISO-8601, redo mapping
+     * @var int[]
+     */
+    private static $intlCalendarMapping = array(
+        \IntlCalendar::DOW_MONDAY    => 1,
+        \IntlCalendar::DOW_TUESDAY   => 2,
+        \IntlCalendar::DOW_WEDNESDAY => 3,
+        \IntlCalendar::DOW_THURSDAY  => 4,
+        \IntlCalendar::DOW_FRIDAY    => 5,
+        \IntlCalendar::DOW_SATURDAY  => 6,
+        \IntlCalendar::DOW_SUNDAY    => 7,
+    );
+
+    const PATTERN_DOW = 'e';
```

Comment:
From looking at http://userguide.icu-project.org/formatparse/datetime and the accepted parameter values I would suggest a different naming pattern for the constants. If you accept global and local patterns this should be reflected in the constant names too. So something like PATTERN_LOCALIZED_SHORT seems to be better since you could also pass PATTERN_GLOBAL_SHORT = 'EEE' or maybe PATTERN_ISO8601_SHORT for example. Make use of the defined constants in L77 too since this is essentially code duplication and introduces more spots to change the same thing.